flashcards/backend/decks/tests/test_models_difficulty_level.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+from django.db import IntegrityError
+
+from decks.models import DifficultyLevel
+
+
+def test_difficulty_level_gui_representation():
+    difficulty_level = DifficultyLevel(name="hard", value=1)
+    assert str(difficulty_level) == difficulty_level.name
+
+
+def test_difficulty_level_in_db(difficulty_level):
+    assert DifficultyLevel.objects.count() == 1
+    assert DifficultyLevel.objects.first().name == difficulty_level.name
+
+
+def test_difficulty_level_meta_information():
+    assert DifficultyLevel._meta.get_field("name").max_length == 20
+    assert DifficultyLevel._meta.get_field("name").unique is True
+    assert DifficultyLevel._meta.get_field("value").unique is True
+    assert DifficultyLevel._meta.verbose_name == "difficulty level"
+    assert DifficultyLevel._meta.verbose_name_plural == "difficulty level`s"
+
+
+def test_difficulty_level_unique_constrain(db, difficulty_level):
+    with pytest.raises(IntegrityError) as excinfo:
```

Comment:
Note for future, if you trigger validators with `obj.full_clean()` you will get a different error a ValidationError and message will be different. You can test if the right message is displayed.